skglm/solvers/group_bcd_solver.py

```diff
@@ -99,19 +115,6 @@ def bcd_solver(X, y, datafit, penalty, w_init=None, p0=10,
                     break
 
         p_obj = datafit.value(y, w, Xw) + penalty.value(w)
-        grad = _construct_grad(X, y, w, Xw, datafit, all_groups)
-        opt = penalty.subdiff_distance(w, grad, all_groups)
-        stop_crit = np.max(opt)
-
-        if verbose:
-            print(
-                f"Iteration {t+1}: {p_obj:.10f}, "
-                f"stopping crit: {stop_crit:.2e}"
-            )
-
-        if stop_crit <= tol:
-            break
-
```

Comment:
I don't understand why you no longer check convergence at the end of one inner loop.

Reply:
I checked the implementation of ``cd_solver`` as well as ``skglm`` paper and I find out that we only do the check at the beginning.